substrate/frame/contracts/src/lib.rs

```diff
@@ -411,8 +411,9 @@ pub mod pallet {
 			use migration::MigrateResult::*;
 
 			loop {
-				let (result, weight) = Migration::<T>::migrate(remaining_weight);
-				remaining_weight.saturating_reduce(weight);
+				const reduced_weight_factor: u64 = 4;
+				let (result, weight) = Migration::<T>::migrate(remaining_weight.saturating_div(reduced_weight_factor));
+				remaining_weight.saturating_reduce(weight.saturating_mul(reduced_weight_factor));
```

Comment:
this [`saturating_reduce`](https://github.com/Cardinal-Cryptography/polkadot-sdk/blob/d8d90a82a7a81edc0f3a36a94fbdbd1602a9e31d/substrate/primitives/arithmetic/src/traits.rs#L289-L298) is weird xD it requires `One` and it's suspicious that it is more effective that `saturating_sub` (because that's my guess for the only reason for such a function)

Reply:
I think it's actually less effective, it's only that `saturating_sub` outputs a value and `saturating_reduce` mutates self. But why the hell is `saturating_reduce` implemented the way it is I have no idea